### PR TITLE
add env-var resource

### PR DESCRIPTION
**Pull request by twhiston**
W.I.P commit for issue #164 

It still needs some integration tests adding, but I thought I would run it past you first to see if you agree with the direction this is going.

it currently just captures the name and the value, as per the kernal-param implementation, but I suppose it could be extended to capture a state of presence and value, so you could either check for its existence or for an explicit value.

### CODE REVIEW

**Comment by twhiston** [comment]
@aelsabbahy please can you give me some guidance on what you expect in the tests and what else i need to get this PR accepted?
To be honest I find the current (non go test) testing setup a little confusing, and trying to run the tests on a mac is a bit of a non starter as well, so it would be super useful to get a rough outline of what I need to do and the general methodology of the testing. Thanks very much!

**Comment by aelsabbahy** [comment]
Unfortunately the testing for goss isn't the most ideal. Currently the Docker/e2e tests are the main form of testing. There are a few `*_test.go` files, but not for the changes you're adding.

To add tests for this:
1. Update the Dockerfiles if needed
2. Update the integration-tests/goss/**/*.yaml as needed
3. Update the `integration-test/test.sh` with the correct test count

Your current run is failing with:
```
+docker exec goss_int_test_centos7_amd64 /goss/centos7/goss-linux-amd64 --vars /goss/vars.yaml -g /goss/centos7/goss.yaml validate
+out='Error: yaml: unmarshal errors:
  line 189: cannot unmarshal !!bool `true` into map[string]interface {}'
```

See my other comment on where I think the mistake is.

File: integration-tests/goss/goss-shared.yaml [hunk 1]
```diff
@@ -177,3 +177,6 @@ matching:
         - have-key-with-value:
             foo: bar
         - have-key: baz
+env-var:
+  GOSS_TEST: true
```
Comment (aelsabbahy) [inline integration-tests/goss/goss-shared.yaml]: I think this needs to be `GOSS_TEST: "true"` to ensure "true" is a string not a boolean value.

**Comment by timbirk** [comment]
This is of interest to me,

One consideration, Environment variables depend heavily on the process or user that they get passed to. Is there some way to validate that for example my Java app has certain env variables set or not?

Should that be part of my applications healthcheck or something goss could check?

**Comment by berney** [comment]
@timbirk in linux you could check `/proc/$pid/environ`, where `$pid` is the process ID of the process you want to check the environment variables of. The file contains them separated by NULLs (\x00).

**Comment by aelsabbahy** [comment]
Seems this PR has gone stale and never passed testing. Closing.

**Pull request closed without merging**